src/app/containers/main/index.jsx

```diff
@@ -65,12 +67,20 @@ class Main extends Component {
     }
 
     searchRequest(searchTerm, offset) {
+        this.setState(initialState);
```

Comment:
You'll probably want to use the callback method for setState before doing anything else. This will guarantee that the initial state was set before doing any more processing. I still can't believe react doesn't have a promise method for setState.